chainer/gradient_check.py

```diff
@@ -222,16 +226,24 @@ def check_backward(func, x_data, y_grad, params=(),
     # `Variable.backward` method calls `Function.backward` of its creator.
     y[0].backward()
 
+    if dtype is None:
+        cxs = [variable.Variable(x) for x in x_data]
+    else:
+        if len(params) > 0:
+            raise ValueError('`dtype` is available only if `params` is empty')
```

Comment:
Could you tell me why we must forbid the case where dtype is not `None` and `params` is empty?